lua/neo-tree/sources/common/file-items.lua

```diff
@@ -119,7 +119,8 @@ function create_item(context, path, _type)
     end
     if f.hide_by_pattern then
       for _, p in ipairs(f.hide_by_pattern) do
-        if string.find(name, p) then
+        local match = string.find(p, '%/') and path or name
```

Comment:
The only thing I would change is to use `utils.path_separator` so that this supports windows as well.

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/a922c3aff32f3a9498466dd1aa99e14df360fbc5/lua/neo-tree/utils.lua#L513

Reply:
I'll need to spend some more time on this over the weekend.
My quick look the glob function is swallowing up the separators if you pass 'C:\path\file.txt' so will need to work out how best to solve for that